### PR TITLE
3x tweaks

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/BaseQueryContext.java [hunk 1]
```diff
@@ -337,18 +337,36 @@ protected void log(final LogLevel level, final QueryNode node, final String log)
       return;
     }
     
+    final StringBuilder buf = new StringBuilder(15 + 6 + 32 + log.length())
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/BaseQueryContext.java]: These numbers in a vacuum are magical. I know they're meaningful, but, someday, using named constants instead could help with intelligibility.

File: core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java [hunk 2]
```diff
@@ -526,7 +526,6 @@ void processResults() {
           return;
         }
       }
-      return;
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java]: Conspicuous. Why continue when we previously did not?

Comment (manolama) [reply core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java]: Cause it turns out we can have a successful query when we have some cache misses :) I had this in there by accident.

File: core/src/main/java/net/opentsdb/query/BaseQueryContext.java (hunk 1 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/query/BaseQueryContext.java]: Added a note and fixed them. Just estimates of the strings we'll be adding so we pre-build the buffer.

File: core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java [hunk 3]
```diff
@@ -917,30 +966,51 @@ protected boolean checkComplete() {
           (context.query().getCacheMode() == CacheMode.NORMAL ||
            context.query().getCacheMode() == CacheMode.WRITEONLY)) {
         if (full_query_context.cacheable()) {
+          full_query_caching.set(true);
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java]: What consumes this new field?

Comment (manolama) [reply core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java]: actually.... nothing, need to remove it.

File: common/src/main/java/net/opentsdb/query/readcache/QueryReadCache.java [hunk 4]
```diff
@@ -99,5 +99,22 @@ public void fetch(final QueryPipelineContext context,
...
+  /**
+   * Remove the entry from the cache.
+   * @param timestamp The Unix epoch timestamp in seconds.
+   * @param key A non-null and non-empty key.
+   * @return A deferred resolving to null on complete.
+   */
+  public Deferred<Void> delete(final int timestamp, final byte[] key);
+  
+  /**
+   * Removes the entries from the cache.
```
Comment (SeanPMiller) [inline common/src/main/java/net/opentsdb/query/readcache/QueryReadCache.java]: Clarify that timestamps are matched to keys by index.

File: common/src/main/java/net/opentsdb/query/readcache/QueryReadCache.java [hunk 5]
```diff
@@ -99,5 +99,22 @@ public void fetch(final QueryPipelineContext context,
...
+   * @param timestampsAn array of Unix epoch timestamps in seconds, must be
```
Comment (SeanPMiller) [inline common/src/main/java/net/opentsdb/query/readcache/QueryReadCache.java]: timestampsAn <- space here

File: common/src/main/java/net/opentsdb/query/readcache/QueryReadCache.java [hunk 6]
```diff
@@ -99,5 +99,22 @@ public void fetch(final QueryPipelineContext context,
...
+   * the same size as the keys.
+   * @param keys A non-null and non-empty array of non-null and non-empty keys.
+   * @return A deferred resolving to null on complete.
+   */
```
Comment (SeanPMiller) [inline common/src/main/java/net/opentsdb/query/readcache/QueryReadCache.java]: Doesn't this check array length and potentially throw?

**Pull request merged**